packages/api2/src/periodsettings/periodsettings.service.ts

```diff
@@ -0,0 +1,111 @@
+import { ServiceException } from '@/shared/service-exception';
+import { UtilsProvider } from '@/utils/utils.provider';
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Request } from 'express';
+import { Model, Types } from 'mongoose';
+import { PeriodDocument } from 'src/periods/schemas/periods.schema';
+import { PeriodStatusType } from '../../../api/src/period/types';
+import {
+  PeriodSetting,
+  PeriodSettingDocument,
+} from './schemas/periodsettings.schema';
+import { SetPeriodSettingDto } from './dto/set-periodsetting.dto';
+import { UploadedFile } from 'express-fileupload';
+import { PeriodsService } from '../periods/periods.service';
+
+@Injectable()
+export class PeriodSettingsService {
+  constructor(
+    @InjectModel(PeriodSetting.name)
+    private periodSettingsModel: Model<PeriodSettingDocument>,
+    private periodsService: PeriodsService,
+    private utils: UtilsProvider,
+  ) {}
+
+  async findAll(periodId: Types.ObjectId): Promise<PeriodSetting[]> {
+    const period = await this.periodsService.findOneById(periodId);
+    if (!period) throw new ServiceException('Period not found.');
```

Comment:
Since periodsService throws when not finding period we don't need to throw again here.